Yoast/wordpress-seo
# Introduce Migration_Status object to handle migration locks and conditionals

## Summary

This PR can be summarized in the following changelog entry:

* No user-facing changes.

## Relevant technical choices:

* Updates the migrations that are currently still behind the indexables feature flag to perform migrations in a more robust manner.

## Test instructions

This PR can be tested by following these steps:

* Ensure you have a clean setup on which no indexable migrations have been run.
* Visit any page.
* The migrations should run and the classes that create indexables when posts, terms or users are saved should create/update entries in the indexables table.
* Delete all tables created by the migration status including the migrations table itself.
* Recreate one of those tables with whatever columns you want manually without running any of the migrations.
* Visit any page.
* You should not see any fatals and the migration should not run.

## Documentation

## Quality assurance


Fixes #12940.
